rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -144,13 +146,160 @@ void testGetOldFederation(
+
+        StorageAccessor storageAccessor = new InMemoryStorage();
+        byte[] federationFormatSerialized = getFederationFormatSerialized(INVALID_FEDERATION_FORMAT);
+        storageAccessor.saveToRepository(PENDING_FEDERATION_FORMAT_VERSION.getKey(), federationFormatSerialized);
+
+        byte[] serializedFederation = expectedFederation.serialize(activations);
+        storageAccessor.saveToRepository(PENDING_FEDERATION_KEY.getKey(), serializedFederation);
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
+
+        // Act
+
+        PendingFederation obtainedFederation = federationStorageProvider.getPendingFederation();
+
+        // Directly saving a null federation in storage to then assert that the method returns the cached federation
+        storageAccessor.saveToRepository(PENDING_FEDERATION_KEY.getKey(), null);
+
+        // Assert
+
+        // Call the method again and assert the same cached federation is returned
+        assertEquals(obtainedFederation, federationStorageProvider.getPendingFederation());
+
+
+        assertArrayEquals(expectedFederation.serialize(activations), obtainedFederation.serialize(activations));
+
+    }
+
+    @Test
+    void getPendingFederation_previouslySet_returnsCachedPendingFederation() {
+
+        // Arrange
+
+        PendingFederation expectedPendingFederation = new PendingFederationBuilder().build();
+
+        // Act
+
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(null);
+        federationStorageProvider.setPendingFederation(expectedPendingFederation);
+
+        // Assert
+
+        assertEquals(expectedPendingFederation, federationStorageProvider.getPendingFederation());
+
+    }
+
+    @Test
+    void getPendingFederation_previouslySetToNull_returnsNull() {
+        // Arrange
+
+        StorageAccessor storageAccessor = mock(StorageAccessor.class);
+
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
+
+        // Act
+
+        federationStorageProvider.setPendingFederation(null);
+        PendingFederation pendingFederation = federationStorageProvider.getPendingFederation();
+
+        // Assert
+
+        assertNull(pendingFederation);
+        verify(storageAccessor, never()).getFromRepository(any(), any());
+
+    }
+
+    @Test
+    void getPendingFederation_nullPendingFederationInStorage_returnsNull() {
+
+        // Arrange
+
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+
+        when(activations.isActive(ConsensusRule.RSKIP123)).thenReturn(false);
+
+        StorageAccessor storageAccessor = new InMemoryStorage();
+        byte[] federationFormatSerialized = getFederationFormatSerialized(STANDARD_MULTISIG_FEDERATION_FORMAT_VERSION);
+        storageAccessor.saveToRepository(PENDING_FEDERATION_FORMAT_VERSION.getKey(), federationFormatSerialized);
```

Comment:
rskip123 isn't supposed to be active if you're saving the format version? 
https://github.com/rsksmart/rskj/blob/add-getPendingFederation-set-of-tests/rskj-core/src/main/java/co/rsk/peg/federation/FederationStorageProviderImpl.java#L361-#L364

Reply:
btw, i don't really follow why you activate or deactivate that rskip in each case

Reply:
Good catch, but the `getPendingFederation` doesn't check the rskip to know if it should use the format version or not, it simply tries to get it from storage. If some other method directly saves that format version in storage, then the `getPendingFederation` method will use it, with rskip active or not.

Is this behavior something we should discuss with the team much deeper? This looks like something that can be 'worked around' to make the `getPendingFederation` use the version key even when the rskip is not active.

What do you guys think? @nathanieliov , @marcos-iov , @julia-zack .

Reply:
Should we have a check for the rskip here? https://github.com/rsksmart/rskj/blob/c4402b35b0e536bcdc5936064e51196c51ad3544/rskj-core/src/main/java/co/rsk/peg/federation/FederationStorageProviderImpl.java#L191